test/requests/hover_expectations_test.rb

```diff
@@ -155,6 +155,35 @@ class A
     T.must(message_queue).close
   end
 
+  def test_hovering_over_gemfile_dependency
+    message_queue = Thread::Queue.new
+    store = RubyLsp::Store.new
+
+    uri = URI("file:///Gemfile")
+    source = <<~RUBY
+      gem 'bundler'
```

Comment:
To prevent any regressions, let's also add a few cases for valid Ruby yet incomplete syntax. Stuff like this

```ruby
# missing argument
gem

# Dynamic gem names
gem(method_call)
```